Cookbook/Technical-Documents/DebugBuilder.md

```diff
@@ -0,0 +1,96 @@
+Debug Builder
+======================
+
+[\[CNSMR\-1602\] Add \`DebugAuthenticatedRootBuilder\` for faster UI debugging iteration by inamiy · Pull Request \#7645 · Babylonpartners/babylon\-ios](https://github.com/Babylonpartners/babylon-ios/pull/7645)
+
+Above PR introduces `DebugAuthenticatedRootBuilder` that allows to directly show a target view builder after authentication.
+
+This feature improves overall UI debugging iteration speed.
+
+## How to make a debug-builder
+
+1. Add a new case in `enum DebugBuilderPattern`
+2. Create a (mock) builder in `DebugAuthenticatedRootBuilder`
+3. Update `Settings.bundle/Root.plist` by adding a new case key
+4. Set `Settings.bundle`'s userDefaults flag via `Settings.app`
+5. Launch the app
+
+## Example
+
+`LocalFeatureSwitches.swift`
+
+```diff
+ public enum DebugBuilderPattern: String {
+     case none
+     ...
+
++    // NOTE: Add a new debug builder case here.
++    case eligibilityCheckBuilder
+ }
+```
+
+`DebugAuthenticatedRootBuilder.swift`
+
+```diff
+ final class DebugAuthenticatedRootBuilder {
+     private let dependencies: AppDependencies
+     private let syncOrchestrator: SyncOrchestrator
+
+     ...
+
+     func make(
+         session: MainUserSession,
+         authenticatedPushEvents: Signal<PushEvent, NoError>,
+         authenticatedApplicationInvocationEvents: Signal<ApplicationInvocation, NoError>,
+         lifecycleEvents: Signal<LifecycleEvent, NoError>
+     ) -> UIViewController? {
+
+         switch LocalFeatureSwitches.debugBuilder {
+         ...
+
++        // NOTE: Implement a new mock builder here.
++        case .eligibilityCheckBuilder:
++            let builder = EligibilityCheckBuilder(dependencies: dependencies, session: session)
++            return DesignLibrary.viewControllersBuilder().navigationController(dismissalStyle: .pop) { (navigation, modal) -> UIViewController in
++                return builder.make(
++                    navigation: navigation,
++                    modal: modal,
++                    presenting: modal,
++                    root: modal,
++                    switchFlow: NHSSwitchFlow(
++                        flow: modal,
++                        rootViewController: UIApplication.shared.delegate?.window??.rootViewController!
++                    )
++                )
++            }
++    }
+ }
+```
+
+`Settings.bundle/Root.plist`
```

Comment:
```suggestion
`Settings.bundle/Root.plist` AND `Babylon/Brand/babylon/Settings.bundle/Root.plist`
```

Reply:
Writing a specific path is easy to get changed, so I decided to add a note comment instead in 2a1b5da .